### PR TITLE
🐛Don't reboot if HetzerBareMetalMachine not found

### PR DESCRIPTION
**Pull request by aniruddha2000**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: pkg/services/baremetal/remediation/remediation.go [hunk 1]
```diff
@@ -53,6 +54,15 @@ func NewService(scope *scope.BareMetalRemediationScope) *Service {
 func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err error) {
 	host, err := s.getHost(ctx)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = fmt.Errorf("HetzerBareMetalMachine not found")
+			if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
+				err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
+				record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
+				return res, err
```
Comment (janiskemper) [inline pkg/services/baremetal/remediation/remediation.go:62]: you don't want to return an error in this case I guess. You did an action (-> manipulated the machine object) and that's as expected.

**Review by janiskemper — CHANGES_REQUESTED** [review]
can you also return in line 79? I think we did a mistake there to not return after manipulating the machine

File: pkg/services/baremetal/remediation/remediation.go [hunk 2]
```diff
@@ -59,9 +59,7 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 			if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
 				err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
 				record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
-				return res, err
 			}
-			return res, err
```
Comment (janiskemper) [inline pkg/services/baremetal/remediation/remediation.go]: I think there was a misunderstanding.. I said do not return an error, not do not return at all :D Please keep the return here!

File: pkg/services/baremetal/remediation/remediation.go [hunk 3]
```diff
@@ -59,9 +59,7 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 			if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
 				err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
 				record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
-				return res, err
```
Comment (janiskemper) [inline pkg/services/baremetal/remediation/remediation.go]: I actually think that here we should still return an error even, as this error that we couldn't update the Machine object is unexpected

**Review by janiskemper — CHANGES_REQUESTED** [review]

**Review by janiskemper — APPROVED** [review]

**Comment by aniruddha2000** [comment]
yes

**Pull request merged**